### PR TITLE
added support to version 11.1

### PR DESCRIPTION
**Pull request by yaakovi**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/13683

## Description
* added support to version 11.1: lines 291-294.
* fixed error handling messages.
* added main function.

## Screenshots
Paste here any images that will help the reviewer

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

File: Integrations/McAfee_ESM-v10/McAfee_ESM-v10.py [hunk 1]
```diff
@@ -161,27 +161,27 @@ def logout(self):
...
+                               headers=self.session_headers,
+                               data=query, verify=VERIFY)
         if not no_validation:
             if no_answer:
-                if self.result.status_code != 200:
-                    return_error(
-                        'Error - ESM replied with:\n - status code: {} \n - body: {}'.format(self.result.status_code,
-                                                                                             self.result.text))
+                if result.status_code != 200:
+                    raise RuntimeError(
```
Comment (anara123) [inline Integrations/McAfee_ESM-v10/McAfee_ESM-v10.py]: why RuntimeError and not ValueError?

Comment (yaakovi) [reply Integrations/McAfee_ESM-v10/McAfee_ESM-v10.py]: no reason, will update

File: Integrations/McAfee_ESM-v10/McAfee_ESM-v10.py [hunk 2]
```diff
@@ -288,7 +288,10 @@ def fetch_alarms(self, time_range, custom_start, custom_end, assigned_user):
         if time_range == 'CUSTOM' and (not custom_start or not custom_end):
             return_error('you must specify customStart and customEnd when timeRange is CUSTOM')
 
-        cmd = 'alarmGetTriggeredAlarmsPaged?pageSize=500&pageNumber=1&triggeredTimeRange=' + time_range
+        if VERSION.startswith('11.'):
+            cmd = 'alarmGetTriggeredAlarms?pageSize=500&pageNumber=1&triggeredTimeRange=' + time_range
```
Comment (anara123) [inline Integrations/McAfee_ESM-v10/McAfee_ESM-v10.py]: change the pageSize to 50
Make sure the results are sorted in ascending order.

File: Integrations/McAfee_ESM-v10/McAfee_ESM-v10.py [hunk 3]
```diff
@@ -207,38 +208,38 @@ def execute_query(self, time_range, custom_start, custom_end, filters, fields, q
...
         for i in range(max_wait):
-            ready = self.get_query_result(resultID)
+            ready = self.get_query_result(result_id)
             if ready:
                 return
             else:
                 time.sleep(60)
 
-        return_error('Waited more {} min for query results : {}'.format(max_wait, resultID))
+        return_error('Waited more {} min for query results : {}'.format(max_wait, result_id))
```
Comment (anara123) [inline Integrations/McAfee_ESM-v10/McAfee_ESM-v10.py]: raise ValueError instead

File: Integrations/McAfee_ESM-v10/McAfee_ESM-v10.py [hunk 4]
```diff
@@ -288,9 +289,21 @@ def fetch_alarms(self, time_range, custom_start, custom_end, assigned_user):
         if time_range == 'CUSTOM' and (not custom_start or not custom_end):
             return_error('you must specify customStart and customEnd when timeRange is CUSTOM')
```
Comment (anara123) [inline Integrations/McAfee_ESM-v10/McAfee_ESM-v10.py]: raise ValueError instead return_error - return_error should be done on Exception catch

**Comment by anara123** [comment]
@yaakovi merging?

**Pull request merged**